rclc/test/rclc/test_executor.cpp

```diff
@@ -728,6 +728,169 @@ TEST_F(TestDefaultExecutor, executor_add_subscription_too_many) {
+    "number of subscriptions is expected to be one";
+
+  // test: remove subscription
+  rc = rclc_executor_remove_subscription(&executor, &this->sub1);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  num_subscriptions = 0;
+  EXPECT_EQ(executor.info.number_of_subscriptions, num_subscriptions) <<
+    "number of subscriptions is expected to be zero";
+
+  // test: remove non-existant should error
+  rc = rclc_executor_remove_subscription(&executor, &this->sub1);
+  EXPECT_EQ(RCL_RET_ERROR, rc) << rcl_get_error_string().str;
+  rcutils_reset_error();
+  EXPECT_EQ(executor.info.number_of_subscriptions, num_subscriptions) <<
+    "number of subscriptions is expected to be zero";
+
+  // test: remove from NULL executor should error
+  rc = rclc_executor_remove_subscription(NULL, &this->sub1);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rc) << rcl_get_error_string().str;
+  rcutils_reset_error();
+  // test: remove null subscription should error
+  rc = rclc_executor_remove_subscription(&executor, NULL);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rc) << rcl_get_error_string().str;
+  rcutils_reset_error();
+
+  // test: add first subscription again
+  rc = rclc_executor_add_subscription(
+    &executor, &this->sub1, &this->sub1_msg,
+    &CALLBACK_1, ON_NEW_DATA);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  num_subscriptions = 1;
+  EXPECT_EQ(executor.info.number_of_subscriptions, num_subscriptions) <<
+    "number of subscriptions is expected to be one";
+
+  // test: add second subscription
+  rc = rclc_executor_add_subscription(
+    &executor, &this->sub2, &this->sub2_msg,
+    &CALLBACK_1, ON_NEW_DATA);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  num_subscriptions = 2;
+  EXPECT_EQ(executor.info.number_of_subscriptions, num_subscriptions) <<
+    "number of subscriptions is expected to be two";
+
+  // test: add third subscription - execution order is 1,2,3
+  rc = rclc_executor_add_subscription(
+    &executor, &this->sub3, &this->sub3_msg,
+    &CALLBACK_1, ON_NEW_DATA);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  num_subscriptions = 3;
+  EXPECT_EQ(executor.info.number_of_subscriptions, num_subscriptions) <<
+    "number of subscriptions is expected to be three";
+
+  // test: the handles are in the expected order
+  EXPECT_EQ(executor.handles[0].subscription, &this->sub1) <<
+    "expect to find sub1 in first handle";
+  EXPECT_EQ(executor.handles[1].subscription, &this->sub2) <<
+    "expect to find sub2 in second handle";
+  EXPECT_EQ(executor.handles[2].subscription, &this->sub3) <<
+    "expect to find sub3 in third handle";
+
+  // test: remove last handle
+  rc = rclc_executor_remove_subscription(&executor, &this->sub3);
+  EXPECT_EQ(RCL_RET_OK, rc) << rcl_get_error_string().str;
+  num_subscriptions = 2;
+  EXPECT_EQ(executor.info.number_of_subscriptions, num_subscriptions) <<
+    "number of subscriptions is expected to be two";
+
```

Comment:
```suggestion
  EXPECT_EQ(executor.handles[0].subscription, &this->sub1) <<
    "expect to find sub1 in first handle";
  EXPECT_EQ(executor.handles[1].subscription, &this->sub2) <<
    "expect to find sub2 in second handle";
```

Reply:
I added this test directly after the remove-operation folowing the principle: one method-call (here `rclc_executor_remove_subscription`) then test expected value with actual value. I saw, of course, that you test the same after adding the subscription again at the end.